File: client/models/Expense.ts
export interface Expense {
  expenseId: number;
  roomId: number;
  payerMembershipId: number;
  payerName: string;
  amount: number;
  description: string;
  category?: string;
  expenseDate: string;
  receiptUrl?: string;
  createdAt: string;
  splits: ExpenseSplit[];
}

export interface ExpenseSplit {
  splitId: number;
  expenseId: number;
  membershipId: number;
  amountOwed: number;
  isPaid: boolean;
  paidAt?: string;
  memberName: string;
}

export interface ExpenseCreateRequest {
  roomId: number;
  payerMembershipId: number;
  amount: number;
  description: string;
  category?: string;
  expenseDate: string;
  receiptUrl?: string;
  splitWith: number[];
}

export interface ExpenseUpdateRequest {
  expenseId: number;
  amount?: number;
  description?: string;
  category?: string;
  expenseDate?: string;
  receiptUrl?: string;
  splitWith?: number[];
}

export interface ExpensePaymentRequest {
  splitId: number;
  membershipId: number;
}

export interface ExpenseSummary {
  totalOwed: number;
  totalOwedToUser: number;
  netBalance: number;
}

export const EXPENSE_CATEGORIES = [
  "Groceries",
  "Utilities",
  "Rent",
  "Maintenance",
  "Cleaning Supplies",
  "Internet",
  "Cable/TV",
  "Furniture",
  "Appliances",
  "Other",
] as const;

export type ExpenseCategory = (typeof EXPENSE_CATEGORIES)[number];

File: api/src/models/expense.py
from datetime import date, datetime
from pydantic import BaseModel
from typing import Optional, List
from decimal import Decimal

class ExpenseCreateRequest(BaseModel):
    room_id: int
    payer_membership_id: int
    amount: Decimal
    description: str
    category: Optional[str] = None
    expense_date: date
    receipt_url: Optional[str] = None
    split_with: List[int]

class ExpenseUpdateRequest(BaseModel):
    expense_id: int
    room_id: int
    payer_membership_id: int
    amount: Decimal
    description: str
    category: Optional[str] = None
    expense_date: date
    receipt_url: Optional[str] = None
    split_with: List[int]

class Expense(BaseModel):
    expense_id: int
    room_id: int
    payer_membership_id: int
    amount: Decimal
    description: str
    category: Optional[str] = None
    expense_date: date
    receipt_url: Optional[str] = None
    created_at: datetime

class ExpenseSplit(BaseModel):
    split_id: int
    expense_id: int
    membership_id: int
    amount_owed: Decimal
    is_paid: bool = False
    paid_at: Optional[datetime] = None

class ExpenseWithSplits(BaseModel):
    expense_id: int
    room_id: int
    payer_membership_id: int
    payer_name: str
    amount: Decimal
    description: str
    category: Optional[str] = None
    expense_date: date
    receipt_url: Optional[str] = None
    created_at: datetime
    splits: List[ExpenseSplit]
    
class ExpensePaymentRequest(BaseModel):
    split_id: int
    membership_id: int
